e2e_tests/e2e_writer.go

```diff
@@ -0,0 +1,614 @@
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/grafana/loki/pkg/push"
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/prometheus/prompb"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	"go.opentelemetry.io/otel/trace"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+var (
+	Shard string
+	//ExtraHeaders     map[string]string
+
+	writingCompleted bool
+	orderMutex       sync.Mutex
+)
+
+func init() {
+	// Initialize variables
+
+	Shard = "default"
+	//	initExtraHeaders()
+	//storage = make(map[string]interface{})
+}
+
+// Global variables to track test execution order
+var (
+	executionOrder []string
+)
+
+// recordExecution records the order of test execution
+func recordExecution(testName string) {
+	orderMutex.Lock()
+	defer orderMutex.Unlock()
+	executionOrder = append(executionOrder, testName)
+	fmt.Printf("Executed test: %s\n", testName)
+}
+
+// Main test suite definition using Ginkgo's Ordered to enforce sequential execution between suites
+func writingTests() {
+	// WritingTests suite runs first
+	Context("Writing Tests", func() {
+		// Verify that writingCompleted is false at the beginning of WritingTests
+		BeforeEach(func() {
+			Expect(writingCompleted).To(BeFalse(), "Writing tests should run before being marked as completed")
+		})
+		It("push logs http", func(ctx context.Context) {
+			testName := "push-logs-http"
+			recordExecution(testName)
+
+			Points := CreatePoints(testID+"_json", 1, start, end, map[string]string{}, nil, nil, nil)
+			Points = CreatePoints(testID, 2, start, end, map[string]string{}, nil, nil, nil)
+			Points = CreatePoints(testID, 3, start, end, map[string]string{}, nil, nil, nil)
+			Points = CreatePoints(testID, 4, start, end, map[string]string{}, nil, nil, nil)
+
+			// JSON format logs
+			Points = CreatePoints(testID+"_json", 1, start, end,
+				map[string]string{"fmt": "json", "lbl_repl": "val_repl", "int_lbl": "1"},
+				Points,
+				func(i int) string {
+					return fmt.Sprintf(`{"lbl_repl":"REPL","int_val":"1","new_lbl":"new_val","str_id":%d,"arr":[1,2,3],"obj":{"o_1":"v_1"}}`, i)
+				},
+				nil,
+			)
+
+			Points = CreatePoints(testID+"_logfmt", 1, start, end,
+				map[string]string{"fmt": "logfmt", "lbl_repl": "val_repl", "int_lbl": "1"},
+				Points,
+				func(i int) string {
+					return fmt.Sprintf(`lbl_repl="REPL" int_val=1 new_lbl="new_val" str_id="%d"`, i)
+				},
+				nil,
+			)
+
+			resp, err := SendPoints(fmt.Sprintf("http://%s", gigaPipeWriteUrl), Points)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.StatusCode).To(BeNumerically("<", 300))
```

Comment:
Different `Expect(resp.StatusCode)` expressions among tests.
The expression should be the same.